### PR TITLE
Always download latest OpenSSL version

### PR DESCRIPTION
**Pull request by quentinmit**
## Description
OpenSSL only makes the latest release available for download. So... just always install the latest version.

## Related issue
b/237004124

## How has this been tested?

Manual build (will rely on CI to test all the combinations of OS)

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [x] Integration tests do not apply.
  - [ ] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.

### CODE REVIEW

**Review by qingling128 — APPROVED** [review]
LGTM modulo the integration tests (aka the agent builds) pass.

**Comment by quentinmit** [comment]
`TestWindowsEventLog/windows-2012-r2` has failed 2/2 times but I have a hard time believing that has anything to do with the openssl version. Trying the tests a third time...

**Comment by quentinmit** [comment]
All of the test failures have been flaky tests that don't appear to relate to OpenSSL at all, and every test has passed at least once, so I'm going to merge this.

**Pull request merged**
